test/graphql/resolvers/collection_test.rb

```diff
@@ -6,6 +6,23 @@ class GraphQLCollectionTest < ActiveSupport::TestCase
   setup do
     @user = FactoryBot.create(:user)
     User.current = @user
+
+    compliance_admin = RBACApiClient::AccessPagination.new(
+      data: [
+        RBACApiClient::Access.new(
+          permission: Rbac::COMPLIANCE_ADMIN,
+          resource_definitions: nil
+        ),
+        RBACApiClient::Access.new(
+          permission: Rbac::INVENTORY_VIEWER,
+          resource_definitions: nil
+        )
+      ]
+    )
+    RBACApiClient::AccessApi
+      .any_instance
```

Comment:
You don't need to stub the `any_instance` because the specific instance is stored in the `Rbac::API_CLIENT` constant. Just stub that one and it'll be a bit faster...

Reply:
The suggestion does not work:

```diff
diff --git a/test/graphql/resolvers/collection_test.rb b/test/graphql/resolvers/collection_test.rb
index d87cc15..9eb4625 100644
--- a/test/graphql/resolvers/collection_test.rb
+++ b/test/graphql/resolvers/collection_test.rb
@@ -20,7 +20,6 @@ class GraphQLCollectionTest < ActiveSupport::TestCase
       ]
     )
     RBACApiClient::AccessApi
-      .any_instance
       .stubs(:get_principal_access)
       .returns(compliance_admin)
   end
```

After the change, test fails:

```
$ podman-compose exec rails bundle exec rake test TEST=test/graphql/resolvers/collection_test.rb
['podman', '--version', '']
using podman version: 4.0.3
podman exec --interactive --tty --env POSTGRESQL_TEST_DATABASE=compliance_test --env DISABLE_DATABASE_ENVIRONMENT_CHECK=1 --env HOSTNAME=rails --env ACG_CONFIG=/app/devel.json compliance-backend_rails_1 bundle exec rake test TEST=test/graphql/resolvers/collection_test.rb
The input [sslMode] is invalid
The input [sslMode] is invalid
`Redis#exists(key)` will return an Integer by default in redis-rb 4.3. The option to explicitly disable this behaviour via `Redis.exists_returns_integer` will be removed in 5.0. You should use `exists?` instead.
Emptying /app/test/reports
Running 1 tests in a single process (parallelization threshold is 50)
Started with run options --seed 47492

ERROR["test_deterministic_pagination", #<Minitest::Reporters::Suite:0x000055c2914b33f0 @name="GraphQLCollectionTest">, 1.1578375130193308]
 test_deterministic_pagination#GraphQLCollectionTest (1.16s)
Minitest::UnexpectedError:         GraphQL::UnauthorizedError: User is not authorized to access this action.
            app/graphql/schema.rb:23:in `unauthorized_object'
            test/graphql/resolvers/collection_test.rb:53:in `block (2 levels) in <class:GraphQLCollectionTest>'
            test/graphql/resolvers/collection_test.rb:40:in `times'
            test/graphql/resolvers/collection_test.rb:40:in `block in <class:GraphQLCollectionTest>'

  1/1: [==================================================================================================================================================] 100% Time: 00:00:01, Time: 00:00:01
Writing XML reports to /app/test/reports

Finished in 1.16024s
1 tests, 0 assertions, 0 failures, 1 errors, 0 skips
Coverage report generated for Unit Tests to /app/coverage. 1081 / 1729 LOC (62.52%) covered.
exit code: 1
```

Reply:
> You don't need to stub the any_instance because the specific instance is stored in the Rbac::API_CLIENT constant.

```ruby
Rbac::API_CLIENT.stubs()
```